libminifi/src/core/yaml/YamlConfiguration.cpp

```diff
@@ -491,14 +510,18 @@ void YamlConfiguration::parsePortYaml(YAML::Node *portNode,
 
   YAML::Node inputPortsObj = portNode->as<YAML::Node>();
 
-  // generate the random UIID
-  uuid_generate(uuid);
-
-  auto portId = inputPortsObj["id"].as<std::string>();
+  // Check for required fields
+  checkRequiredField(&inputPortsObj, "name");
   auto nameStr = inputPortsObj["name"].as<std::string>();
+  checkRequiredField(&inputPortsObj, "id", "The field 'id' is required for "
+      "the port named '" + nameStr + "' in the YAML Config. If this port "
+      "is specificy an input port for a NiFi Remote Process Group, the port "
+      "id should match the id of of the input port in the NiFi configuration. "
+      "This is a UUID of the format XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX.");
+  auto portId = inputPortsObj["id"].as<std::string>();
   uuid_parse(portId.c_str(), uuid);
 
-  port = new minifi::RemoteProcessorGroupPort(nameStr.c_str(), uuid);
+  port = new minifi::RemoteProcessorGroupPort(nameStr, uuid);
 
   processor = (std::shared_ptr<core::Processor>) port;
```

Comment:
I don't see the cast you are referring to in getRoot. I only see the use of the unique_ptr, which  okay as that is currently the only way to create a unique pointer. What I'm referencing is the explicit cast of a pointer to a shared_ptr object, which could have negative consequences at some point. If this is an endemic issue we can create a ticket, but I didn't see another case of this specific cast. The appropriate thing to do is std::make_shared<minifi::RemoteProcessorGroupPort>(nameStr,uuid);